nakadi-producer/src/main/java/org/zalando/nakadiproducer/eventlog/impl/EventLogRepository.java

```diff
@@ -12,6 +12,12 @@ public interface EventLogRepository {
 
```

Comment:
It just came to my mind that apart from batch-storing the events when producing, we likely also want to batch-delete them from the DB after they are sent out.

```suggestion
    default void delete(Collection<EventLog> eventLogs) {
        eventLogs.forEach(this::delete);
    }

```
(with a more efficient implementation using `DELETE ... WHERE id IN (...)` in the EventLogRepositoryImpl).

The EventTransmissionService then can call this method instead of the current `successfulEvents.forEach(eventLogRepository::delete)`.

Reply:
(But we can also do this in a separate PR, if you want.)

Reply:
The problem was observed by tracing, too. Although it does not slow down the business operations themselves, it actually creates unnecessary load on the database. I filed a separate issue for it (https://github.com/zalando-nakadi/nakadi-producer-spring-boot-starter/issues/159).